### PR TITLE
Revamp the shuttle backend

### PR DESCRIPTION
**Pull request by ZephyrTFA**
It does good things

### CODE REVIEW

**Review by ritorizo — COMMENTED** [review]
Disclaimer : I'm not very familiar with the shuttle code.

I looked trough all of it and it's coherent.
Some map object with the property id were changed to base_id which shouldn't have tough (gonna look into it).

**Comment by ritorizo** [comment]
Made a pr on this branch which should resolve the map issues.

**Comment by tmtmtl30** [comment]
i've probably got some substantial thoughts on this, remind me to review it

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -230,7 +230,8 @@ GLOBAL_LIST_EMPTY(crematoriums)
 	connected = new /obj/structure/tray/c_tray(src)
 	connected.connected = src
 
-/obj/structure/bodycontainer/crematorium/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
+/obj/structure/bodycontainer/crematorium/connect_to_shuttle(mapload, obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
+	. = ..()
 	id = "[REF(port)][id]"
```
Comment (MarkSuckerberg) [inline code/game/objects/structures/morgue.dm:235]: Think you meant to remove this

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 2]
```diff
@@ -128,6 +128,7 @@ we use a hook instead
 
 /datum/gas_mixture/proc/__gasmixture_unregister()
 /datum/gas_mixture/proc/__gasmixture_register()
+/datum/gas_mixture/proc/__auxtools_parse_gas_string()
```
Comment (MarkSuckerberg) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm:131]: almost certain this should be atomised into a different PR

File: code/modules/mapping/map_template.dm [hunk 3]
```diff
@@ -186,9 +206,29 @@
...
+		if(findtext(models[key], "[marker]")) // Yay compile time checks
+			break // This works by assuming there will ever only be one mobile dock in a template at most
+
+	for(var/datum/grid_set/gset as anything in cached_map.gridSets)
+		var/ycrd = gset.ycrd
+		for(var/line in gset.gridLines)
+			var/xcrd = gset.xcrd
+			for(var/j in 1 to length(line) step cached_map.key_len)
+				if(key == copytext(line, j, j + cached_map.key_len))
+					return list(xcrd, ycrd)
```
Comment (MarkSuckerberg) [inline code/modules/mapping/map_template.dm:223]: Pretty sure BYOND lists of numbers need to have a key or else there's weirdness with the ordering (e.g. list("x" = x, "y" = y) might be needed)

File: code/datums/shuttles.dm [hunk 4]
```diff
@@ -76,33 +76,10 @@
...
-					port.height = height
-					port.dwidth = width - port_x_offset
-					port.dheight = height - port_y_offset
-				if(WEST)
-					port.width = height
-					port.height = width
-					port.dwidth = port_y_offset - 1
-					port.dheight = width - port_x_offset
-
-			port.load(src)
```
Comment (MarkSuckerberg) [inline code/datums/shuttles.dm:105]: Sure this won't break the change from https://github.com/shiptest-ss13/Shiptest/pull/42?

File: code/modules/shuttle/shuttle.dm [hunk 5]
```diff
@@ -320,38 +402,14 @@
 		docking_points = null
 		shuttle_areas = null //TODO: This is nowhere near enough to clear references, lol. We need an /atom/proc/disconnect_from_shuttle() proc to clear references.
 		remove_ripples()
-	. = ..()
-
-/obj/docking_port/mobile/Initialize(mapload)
-	. = ..()
-	if(!mapload) // If maploaded, will be called in code\datums\shuttles.dm
-		load()
```
Comment (MarkSuckerberg) [inline code/modules/shuttle/shuttle.dm:328]: See comment about https://github.com/shiptest-ss13/Shiptest/pull/42

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
Definitely a few things need looking over and some testing of some cases

**Comment by ZephyrTFA** [comment]
me when all the maps conflict

**Pull request closed without merging**